### PR TITLE
Refactored Truffle Tests

### PR DESCRIPTION
**Pull request by laskdaf**
Created functions in `utilities.js` for repeating processes.
Cleaned up testing code.
Requesting help for `rootchain_exits.js`. Not sure why the code in the before statement is required for the final test to succeed.

### CODE REVIEW

File: contracts/rootchain/RootChain.sol [hunk 1]
```diff
@@ -21,6 +21,7 @@ contract RootChain is Ownable {
     event Deposit(address depositor, uint256 amount);
     event FinalizedExit(uint priority, address owner, uint256 amount);
     event AddedToBalances(address owner, uint256 amount);
+    event QueueSize(uint256 size);
```
Comment (AdityaSripal) [inline contracts/rootchain/RootChain.sol]: Why do we need this? Is it just for testing?

File: contracts/rootchain/RootChain.sol [hunk 2]
```diff
@@ -314,6 +315,7 @@ contract RootChain is Ownable {
         view
         returns (address, uint256, uint256[3], uint256)
     {
+        emit FinalizedExit(priority, exits[priority].owner, exits[priority].amount);
```
Comment (AdityaSripal) [inline contracts/rootchain/RootChain.sol]: Why are we emitting a FinalizedExit here, this is just a function to get an exit.

File: contracts/rootchain/RootChain.sol (hunk 1 above)
Comment (hamdiallam) [reply contracts/rootchain/RootChain.sol]: lol yes, please remove it

Comment (laskdaf) [reply contracts/rootchain/RootChain.sol]: oops i added a few events while i was testing and forgot to get rid of them. good catch

File: contracts/rootchain/RootChain.sol (hunk 2 above)
Comment (laskdaf) [reply contracts/rootchain/RootChain.sol]: same as above

File: test/rootchain_helpers.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
+let RLP = require('rlp');
+
+/*
+ How to avoid using try/catch blocks with promises' that could fail using async/await
+ - https://blog.grossman.io/how-to-write-async-await-without-try-catch-blocks-in-javascript/
+ */
+let to = function(promise) {
```
Comment (AdityaSripal) [inline test/rootchain_helpers.js]: Possibly change the name of this function

Comment (hamdiallam) [reply test/rootchain_helpers.js]: I think it is fine. Open to changing it though

Comment (laskdaf) [reply test/rootchain_helpers.js]: would `promise` work?

Comment (hamdiallam) [reply test/rootchain_helpers.js]: I vote for keeping it. it's concise

Comment (AdityaSripal) [reply test/rootchain_helpers.js]: is it just used in the context of catching errors? it's concise, but someone who is reading through our tests will have to look at the function definition to figure out what to does.

Ideally, people should read the function and know what it does.

Comment (hamdiallam) [reply test/rootchain_helpers.js]: remove the definition from this file

File: test/depositAndExitTxs.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
+// external libraries
+let RLP = require('rlp');
+let assert = require('chai').assert;
+
+let utilities = require('./utilities.js');
```
Comment (hamdiallam) [inline test/depositAndExitTxs.js]: destructure the functions here

File: test/depositAndExitTxs.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+let RLP = require('rlp');
+let assert = require('chai').assert;
+
+let {
+    catchError,
+    toHex,
+    fastForward,
+    proofForDepositBlock,
+    zeroHashes
+} = require('./utilities.js');let rootchainHelpers = require('./rootchain_helpers.js');
```
Comment (hamdiallam) [inline test/depositAndExitTxs.js]: separate lines

File: test/priorityQueue.js [hunk 6]
```diff
@@ -1,6 +1,14 @@
...
+
+let PriorityQueue = artifacts.require("PriorityQueue");
+
+let {
+    catchError,
+    toHex,
+    fastForward,
+    proofForDepositBlock,
+    zeroHashes
+} = require('./utilities.js');let rootchainHelpers = require('./rootchain_helpers.js');
```
Comment (hamdiallam) [inline test/priorityQueue.js]: same here

**Pull request merged**